File: src/com/Torvald/Terrarum/ConsoleCommand/SetLocale.java
package com.Torvald.Terrarum.ConsoleCommand;

import com.Torvald.Terrarum.LangPack.Lang;
import com.Torvald.Terrarum.Terrarum;

import java.io.IOException;

/**
 * Created by minjaesong on 16-01-25.
 */
public class SetLocale implements ConsoleCommand {
    @Override
    public void execute(String[] args) {
        if (args.length == 2) {
            String prevLocale = Terrarum.gameLocale;
            Terrarum.gameLocale = args[1].toLowerCase();
            try {
                new Lang();
                new Echo().execute("Set locale to '" + Terrarum.gameLocale + "'.");
            }
            catch (IOException e) {
                new Echo().execute("Locale '"
                        + args[1].toLowerCase()
                        + "' does not exist or could not read file."
                );
                Terrarum.gameLocale = prevLocale;
            }
        }
        else {
            printUsage();
        }
    }

    @Override
    public void printUsage() {
        new Echo().execute("Usage: setlocale [locale]");
    }
}

File: src/net/torvald/terrarum/console/SetLocale.kt
package net.torvald.terrarum.console

import net.torvald.terrarum.langpack.Lang
import net.torvald.terrarum.AppLoader

import java.io.IOException

/**
 * Created by minjaesong on 2016-01-25.
 */
internal object SetLocale : ConsoleCommand {
    override fun execute(args: Array<String>) {
        if (args.size == 2) {
            val prevLocale = AppLoader.GAME_LOCALE
            AppLoader.GAME_LOCALE = args[1]
            try {
                Echo("Set locale to '" + AppLoader.GAME_LOCALE + "'.")
            }
            catch (e: IOException) {
                Echo("could not read lang file.")
                AppLoader.GAME_LOCALE = prevLocale
            }

        }
        else if (args.size == 1) {

            Echo("Locales:")

            Lang.languageList.forEach { Echo("--> $it") }
        }
        else {
            printUsage()
        }
    }

    override fun printUsage() {
        Echo("Usage: setlocale [locale]")
    }
}
